signer/src/stacks/api.rs

```diff
@@ -71,6 +73,13 @@ pub enum FeePriority {
 
 /// A trait detailing the interface with the Stacks API and Stacks Nodes.
 pub trait StacksInteract {
+    /// Retrieve a data variable from a contract.
+    fn get_data_var(
+        &mut self,
+        contract_principal: &StacksAddress,
+        contract_name: &ContractName,
+        var_name: &ClarityName,
+    ) -> impl Future<Output = Result<Value, Error>>;
```

Comment:
Hmmm, I think it would be better to leave out the generic `get_data_var` function from the trait and instead have a `get_current_signing_set` function (or whatever name) that does the conversion of a clarity `Value` into a "regular" type for you. This should make testing the behavior a little simpler since you wouldn't need to return a `Value` in the test structs that implement the trait.

Reply:
@djordon yeah that's a good point -- I pushed some changes now to refactor how that works a little, with some slightly improved error handling as well.

Reply:
Oh, good call out. I think what you have here is fine.